### PR TITLE
Run tests in parallel with other jobs

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

**Comment by aelred** [comment]
@fppt even though this has failed, it does appear to work:
https://jenkins.grakn.ai/blue/organizations/jenkins/grakn/detail/PR-2329/1/pipeline

File: Jenkinsfile [hunk 1]
```diff
@@ -96,28 +96,44 @@ def buildGrakn() {
...
-	stage('Run Janus test profile') {
-	    try {
-		sh "mvn clean verify -P janus -U -Djetty.log.level=WARNING -Djetty.log.appender=STDOUT"
-	    } finally {
-		junit "**/TEST*.xml"
-	    }
-	}
-    }
+def shouldRunAllTests() {
+    // TODO bad
```
Comment (fppt) [inline Jenkinsfile]: More obvious highlighting of the badness

File: Jenkinsfile [hunk 2]
```diff
@@ -96,28 +96,44 @@ def buildGrakn() {
...
+def shouldDeployLongRunningInstance() {
+    return env.BRANCH_NAME == 'stable'
 }
 
-//Only run validation master/stable
-if (env.BRANCH_NAME in ['master', 'stable']) {
+// This is a map of jobs to perform in parallel, name -> job closure
+jobs = [
+        tests: {
+            //Run all tests
```
Comment (fppt) [inline Jenkinsfile]: It looks like there is only one test specified to run here. Is that on purpose?

File: Jenkinsfile [hunk 3]
```diff
@@ -96,28 +96,44 @@ def buildGrakn() {
...
+            node {
+                String workspace = pwd()
+                checkout scm
+
+                slackGithub "Janus tests started"
+
+                timeout(120) {
+                    stage('Run Janus test profile') {
+                        try {
+                            sh "mvn clean verify -P janus -U -Djetty.log.level=WARNING -Djetty.log.appender=STDOUT"
```
Comment (fppt) [inline Jenkinsfile]: `verify`. . . I know know why the Jenkins build was failing on my crazy test refactor PR.

File: Jenkinsfile [hunk 4]
```diff
@@ -96,28 +96,44 @@ def buildGrakn() {
...
 
-    slackGithub "Janus tests success", "good"
+def shouldDeployLongRunningInstance() {
+    return env.BRANCH_NAME == 'stable'
 }
 
-//Only run validation master/stable
-if (env.BRANCH_NAME in ['master', 'stable']) {
+// This is a map of jobs to perform in parallel, name -> job closure
+jobs = [
```
Comment (fppt) [inline Jenkinsfile]: Looks like this guys is missing a name. His friend below is defined like `jobs['benchmarks']`. The comment is also asking for a name.

Comment (aelred) [reply Jenkinsfile]: It's called `jobs`, it's just a map. I can define it like this:
```
jobs = []

jobs['tests'] = { ... }
```
if that's clearer?

File: Jenkinsfile (hunk 2 above)
Comment (aelred) [reply Jenkinsfile]: Yes, we add them later with `jobs['benchmarks'] = { ... }` etc.

**Comment by aelred** [comment]
@fppt re-review? 👍

**Comment by aelred** [comment]
oh shit I merged this with the `// TODO: bad` thing D:

**Comment by fppt** [comment]
Ooops. I blame your reviewer

**Pull request merged**